docs/content/sips/010-key-value.md

```diff
@@ -0,0 +1,132 @@
+  // configuring a store with that name in a runtime configuration file
+  // may address this.
+  no-such-store,
+
+  // The store handle provided is not recognized, i.e. it was either never
+  // opened or has been closed.
+  invalid-store,
+
+  // No key-value tuple exists for the specified key in the specified
+  // store.
+  no-such-key,
+
+  // Some implementation-specific error has occurred (e.g. I/O)
+  io(string)
+}
+
+// Open the store with the specified name.
+//
+// If `name` is the empty string, the default store is opened.  Otherwise,
+// `name` must refer to a store defined and configured in a runtime
+// configuration file supplied with the application.
+// `error::no-such-store` will be raised if the `name` is not recognized.
+open: func(name: string) -> expected<store, error>
+
+// Get the value associated with the specified `key` from the specified
+// `store`.
+//
+// `error::invalid-store` will be raised if `store` is not a valid handle
+// to an open store, and `error::no-such-key` will be raised if there is no
+// tuple for `key` in `store`.
+get: func(store: store, key: string) -> expected<list<u8>, error>
+
+// Set the `value` associated with the specified `key` in the specified
+// `store`, overwriting any existing value.
+//
+// `error::invalid-store` will be raised if `store` is not a valid handle
+// to an open store.
+set: func(store: store, key: string, value: list<u8>) -> expected<unit, error>
+
+// Delete the tuple with the specified `key` from the specified `store`.
+//
+// `error::invalid-store` will be raised if `store` is not a valid handle
+// to an open store.  No error is raised if a tuple did not previously
+// exist for `key`.
+delete: func(store: store, key: string) -> expected<unit, error>
+
+// Return whether a tuple exists for the specified `key` in the specified
+// `store`.
+//
+// `error::invalid-store` will be raised if `store` is not a valid handle
+// to an open store.
+exists: func(store: store, key: string) -> expected<bool, error>
+
+// Return a list of all the keys in the specified `store`.
+//
+// `error::invalid-store` will be raised if `store` is not a valid handle
+// to an open store.
+get-keys: func(store: store) -> expected<list<string>, error>
+
+// Close the specified `store`.
+//
+// This has no effect if `store` is not a valid handle to an open store.
+close: func(store: store)
+```
+
+*Note: the pseudo-resource design was inspired by the interface of similar functions in [WASI preview 2](https://github.com/bytecodealliance/preview2-prototyping/blob/d56b8977a2b700432d1f7f84656d542f1d8854b0/wit/wasi.wit#L772-L794).*
+
+#### Implementation requirements
+
+In addition to the above interface, we specify a few additional implementation requirements which guest components may rely on.  At minimum, an conforming implementation must support:
+
+- Keys as large as 256 bytes (UTF-8 encoded)
+- Values as large as 1 megabyte
+- Capacity for 1024 key-value tuples
```

Comment:
This might be slightly confusing when combined with the value size requirement, suggesting a cloud provider must provide 1GB storage. Maybe qualify the whole section "subject to overall database size limits" or something?